packages/serialization/json/src/browser/jsonParseNodeFactory.ts

```diff
@@ -0,0 +1,29 @@
+import { ParseNode, ParseNodeFactory } from "@microsoft/kiota-abstractions";
+
+import { JsonParseNode } from "./../jsonParseNode";
+
+export class JsonParseNodeFactory implements ParseNodeFactory {
+  public getValidContentType(): string {
+    return "application/json";
+  }
+  public getRootParseNode(
+    contentType: string,
+    content: ArrayBuffer
+  ): ParseNode {
+    if (!contentType) {
+      throw new Error("content type cannot be undefined or empty");
+    } else if (this.getValidContentType() !== contentType) {
+      throw new Error(`expected a ${this.getValidContentType()} content type`);
+    }
+    if (!content) {
+      throw new Error("content cannot be undefined of empty");
+    }
```

Comment:
minor perf: make this the first check of the function to avoid calling `getValidContentType()` function and a string comparison when `content` is undefined or empty.